homeassistant/components/sensor/solaredge.py

```diff
@@ -0,0 +1,176 @@
+            return False
+        _LOGGER.debug("Credentials correct and site is active")
+    except KeyError:
+        _LOGGER.debug("Missing details data in solaredge response")
+        return False
+    except (ConnectTimeout, HTTPError):
+        _LOGGER.debug("Could not retrieve details from SolarEdge \
+         Monitoring API")
+        return False
+
+    # Create solaredge data service which will retrieve and update the data.
+    data = SolarEdgeData(hass, api, site_id)
+
+    # Create a new sensor for each sensor type.
+    entities = []
+    for sensor_key in config[CONF_MONITORED_VARIABLES]:
+        sensor = SolarEdgeSensor(platform_name, sensor_key, data)
+        entities.append(sensor)
+
+    add_entities(entities, True)
+
+
+class SolarEdgeSensor(Entity):
+    """Representation of an SolarEdge Monitoring API sensor."""
+
+    def __init__(self, platform_name, sensor_key, data):
+        """Initialize the sensor."""
+        self.platform_name = platform_name
+        self.sensor_key = sensor_key
+        self.data = data
+        self._state = None
+
+        self._json_key = SENSOR_TYPES[self.sensor_key][0]
+        self._name = SENSOR_TYPES[self.sensor_key][1]
+        self._unit_of_measurement = SENSOR_TYPES[self.sensor_key][2]
+
+    @property
+    def name(self):
+        """Return the name."""
+        return "{}_{}".format(self.platform_name, self.sensor_key)
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the sensor icon."""
+        return SENSOR_TYPES[self.sensor_key][3]
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    async def async_update(self):
+        """Get the latest data from the sensor and update the state."""
+        _LOGGER.debug("async_update")
+        await self.hass.async_add_job(self.data.update)
+        self._state = self.data.data[self._json_key]
+
+
+class SolarEdgeData:
+    """Get and update the latest data."""
+
+    def __init__(self, hass, api, site_id):
+        """Initialize the data object."""
+        self.hass = hass
+        self.api = api
+        self.data = {}
+        self.site_id = site_id
+
+        self.update()
+
+    @Throttle(UPDATE_DELAY)
+    def update(self):
+        """Update the data from the SolarEdge Monitoring API."""
+        from requests.exceptions import HTTPError, ConnectTimeout
+
+        try:
+            data = self.api.get_overview(self.site_id)
+            overview = data['overview']
+        except KeyError:
+            _LOGGER.debug("Missing overview data, skipping update")
+            return
+        except (ConnectTimeout, HTTPError):
+            _LOGGER.debug("Could not retrieve data, skipping update")
+            return
+
+        self.data = {}
+
+        for item in overview:
```

Comment:
If `overview` is a dict we can do:
```suggestion
        for key, value in overview.items():
```